### PR TITLE
Database/Protocol parameter for API's

### PR DESCRIPTION
**Pull request by prakriti-solankey**

### CODE REVIEW

File: backend/score.py [hunk 1]
```diff
@@ -129,11 +130,25 @@ async def extract_knowledge_graph_from_file(
...
+                          password:str):
     """
     Calls 'get_source_list_from_graph' which returns list of sources which alreday exist in databse
     """
-    result = await asyncio.to_thread(get_source_list_from_graph,uri,database,userName,password)
+    # base64_bytes = base64.b64encode(password.encode("utf-8"))
+    # base64_encoded_string = base64_bytes.decode("utf-8")
+    # print("encoded password=",base64_encoded_string)
+   
+    #base64_bytes = password.encode("ascii")
```
Comment (jexp) [inline backend/score.py]: this should probably handled in one place not in every function

File: backend/src/main.py [hunk 2]
```diff
@@ -190,6 +190,7 @@ def create_source_node_graph_url(uri, db_name, userName, password, source_url, m
            job_status = "Completed"
            return create_api_response(job_status,data='Unknown URL')
     except Exception as e:
+        source_type = ''
         job_status = "Failed"
         error_message = str(e)
         logging.exception(f'Exception Stack trace:')
```
Comment (jexp) [inline backend/src/main.py:199]: exception is not actually logged or exception message returned

File: backend/src/main.py [hunk 3]
```diff
@@ -190,6 +190,7 @@ def create_source_node_graph_url(uri, db_name, userName, password, source_url, m
            job_status = "Completed"
            return create_api_response(job_status,data='Unknown URL')
     except Exception as e:
+        source_type = ''
```
Comment (jexp) [inline backend/src/main.py]: why would we empty out the source type?

**Comment by jexp** [comment]
This PR has a lot of deletions, I guess from some intermediate state?
Can we have a clean baseline PR so that review becomes easier?

**Comment by prakriti-solankey** [comment]
Sure @jexp will clean this up and share for review

File: backend/score.py (hunk 1 above)
Comment (praveshkumar1988) [reply backend/score.py]: Created separate function for decode password

File: backend/src/main.py (hunk 2 above)
Comment (praveshkumar1988) [reply backend/src/main.py:199]: @jexp Exception logging in each function

File: backend/src/main.py (hunk 3 above)
Comment (praveshkumar1988) [reply backend/src/main.py]: @jexp remove empty source_type and handle the exception.

**Review by praveshkumar1988 — APPROVED** [review]
Approved

**Pull request merged**